src/crypto.rs

```diff
@@ -16,36 +18,36 @@ macro_rules! to_enc_error {
     };
 }
 
-fn generichash_quick(msg: &[u8], key: Option<&[u8]>) -> Result<Vec<u8>> {
+fn generichash_quick(msg: &[u8], key: Option<&[u8]>) -> Result<[u8; 32]> {
     let mut state = to_enc_error!(
         generichash::State::new(Some(32), key),
         "Failed to init hash"
     )?;
     to_enc_error!(state.update(msg), "Failed to update hash")?;
     Ok(to_enc_error!(state.finalize(), "Failed to finalize hash")?
         .as_ref()
-        .to_vec())
+        .try_into()
+        .expect("generichash returned result of wrong size"))
```

Comment:
Expect panics, we want to return an error, no?

Reply:
I suppose we can return an error, but there isn't really anything that can be done about this error anyway, it's a violation of the libsodium API guarantees. If we get a wrong size here, something is extremely broken and we should probably just bail.

Reply:
Oh, it comes from the generic hash, yeah, it should be fine.